packages/shorebird_cli/lib/src/executables/patch_executable.dart

```diff
@@ -49,9 +50,20 @@ class PatchExecutable {
     final result = await process.run(diffExecutable, diffArguments);
 
     if (result.exitCode != ExitCode.success.code) {
+      var messageDetails = '';
+
+      if (result.exitCode == -1073741515 && platform.isWindows) {
+        messageDetails = '''
+This indicates that the Microsoft C++ runtime (VCRUNTIME140.dll) could not be found.
+
+The C++ Runtime can be installed from microsoft at:
+${link(uri: Uri.parse('https://www.microsoft.com/en-us/download/details.aspx?id=52685'))}
+''';
+      }
```

Comment:
Minor but might be nice to pull this check out above the generic result.exitCode != ExitCode.success.code to make it reduce the nesting and make it easier to reason about.

```dart
const vcruntimeNotFound = -1073741515;
// Explanation of this special case with a link to the relevant GitHub issue.
if (result.exitCode == vcruntimeNotFound && platform.isWindows) {
  // Specific exception
  throw PatchFailedException(...);
}

if (result.exitCode != ExitCode.success.code) {
  // Generic exception
}
```